elastic/apm-server
# Configuration for dropping non-sampled transactions

## Motivation/summary

Introduce configuration to drop non-sampled transactions. Typically this should only be used in conjunction with transaction aggregation. Off by default for backwards compatibility.

TODO:

~- [ ] Docs~ (will document later, when we have UI support)
- [x] Log warning if aggregation is not enabled
- [x] Monitoring -- increment counter if non-sampled transaction docs are dropped

## Checklist

~- [ ] I have made corresponding changes to the documentation~

## How to test these changes

1. Configure an agent with <100% sampling rate
2. Configure server with `apm-server.sampling.keep_non_sampled=false`
3. Create some transactions
4. Observe that only sampled transactions are recorded

## Related issues

Relates to https://github.com/elastic/apm-server/issues/3485
